csharp/src/Ice/Communicator.cs

```diff
@@ -502,25 +504,24 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
 
                 ServerName = GetProperty("Ice.ServerName") ?? Dns.GetHostName();
 
-                SlicOptions = new SlicOptions
+                MaxBidirectionalStreams = GetPropertyAsInt("Ice.MaxBidirectionalStreams") ?? 100;
+                if (MaxBidirectionalStreams < 1)
                 {
-                    MaxBidirectionalStreams = GetPropertyAsInt("Ice.Slic.MaxBidirectionalStreams") ?? 100,
-                    MaxUnidirectionalStreams = GetPropertyAsInt("Ice.Slic.MaxUnidirectionalStreams") ?? 100,
-                    PacketMaxSize = GetPropertyAsInt("Ice.Slic.PacketMaxSize") ?? 32 * 1024
-                };
-                if (SlicOptions.MaxBidirectionalStreams < 1)
-                {
-                    throw new InvalidConfigurationException($"{SlicOptions.MaxBidirectionalStreams} is not a valid " +
-                        "value for Ice.Slic.MaxBidirectionalStreams");
+                    throw new InvalidConfigurationException($"{MaxBidirectionalStreams} is not a valid " +
+                        "value for Ice.MaxBidirectionalStreams");
                 }
-                if (SlicOptions.MaxUnidirectionalStreams < 1)
+
+                MaxUnidirectionalStreams = GetPropertyAsInt("Ice.MaxUnidirectionalStreams") ?? 100;
+                if (MaxUnidirectionalStreams < 1)
                 {
-                    throw new InvalidConfigurationException($"{SlicOptions.MaxBidirectionalStreams} is not a valid " +
-                        "value for Ice.Slic.MaxUnidirectionalStreams");
+                    throw new InvalidConfigurationException($"{MaxBidirectionalStreams} is not a valid " +
+                        "value for Ice.MaxUnidirectionalStreams");
                 }
-                if (SlicOptions.PacketMaxSize < 1024)
+
+                SlicPacketMaxSize = GetPropertyAsInt("Ice.Slic.PacketMaxSize") ?? 32 * 1024;
+                if (SlicPacketMaxSize < 1024)
```

Comment:
Specifying something smaller doesn't seem very useful to me... Quic uses 1200 for its packet size to limit fragmentation  of the UDP datagram. Slic is based on TCP and the idea of doing the fragmentation at the Slic level with packets of 1KB could be to avoid TCP fragmentation... however, I don't know if this is really useful either as I didn't really do any performance testing or investigate in which use cases this could be beneficial. Do you perhaps have another suggestion for this limit and the default? I think we should eventually run some performance testing for the default value.

Reply:
`1024` seems good to me!